cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java

```diff
@@ -360,6 +362,8 @@ protected void configure() {
     messagingContext = new MultiThreadMessagingContext(messagingService);
     firstInit = false;
     previewManager = injector.getInstance(PreviewManager.class);
+    programLifecycleService = injector.getInstance(ProgramLifecycleService.class);
```

Comment:
better to move this before starting the scheduler, even though currently the scheduler doesn't block to wait for ProgramLifecycleService to start. BTW, what is the purpose of calling `startAndWait()` on ProgramLifecycleService? It seems to be a no-op

Reply:
If `ProgramLifecycleService` actually has nothing start/stop, it shouldn't implement `Service` at all. It's good to clean that up too.

Reply:
After the next PR, it will actually be a Service, as it will subscribe to TMS to start programs once their cluster is provisioned.